### ISSUE TITLE
cluster.vq.whiten does not implement whitening

### ISSUE THREAD

**Issue by rgommers** [scipy/scipy#12453]
Follow-up to gh-12446. From @JesseLivezey:

> An unrelated reason to change the name is that, as far as I can tell, this function does not implement whitening. It looks like it normalizes the features (in the sense of per-feature standard deviation). "Whitening" typically means scaling the singular values of the feature matrix (or eigenvalues of the covariance) to all be 1. This flat eigenvalue spectrum is where the analogy to flattening the power spectrum comes from.
>
> For instance, this is what is done in sklearn's PCA. Just rescaling the features, which is what is happening here, is not whitening. 

This point seems correct to me. There's lots of sources for this, just picking the one from Wikipedia here:

_A whitening transformation or sphering transformation is a linear transformation that transforms a vector of random variables with a known covariance matrix into a set of new variables whose covariance is the identity matrix, meaning that they are uncorrelated and each have variance 1.[1] The transformation is called "whitening" because it changes the input vector into a white noise vector._

_Several other transformations are closely related to whitening:_

_1. the decorrelation transform removes only the correlations but leaves variances intact,
2. the standardization transform sets variances to 1 but leaves correlations intact,
3. a coloring transformation transforms a vector of white random variables into a random vector with a specified covariance matrix._

In that terminology, `cluster.vq.whiten` actually does `standardization`. 

Here's the scikit-learn code referenced: https://github.com/scikit-learn/scikit-learn/blob/2f26540ee99cb4519d7471933359913c7be36ac9/sklearn/decomposition/_pca.py#L159. It says:
```
        When True (False by default) the `components_` vectors are multiplied
        by the square root of n_samples and then divided by the singular values
        to ensure uncorrelated outputs with unit component-wise variances.
```

The scikit-learn `[kmeans implementation](https://github.com/scikit-learn/scikit-learn/blob/c29092d6994a43ea19a82e91c0331e8b7d6e7d36/sklearn/cluster/_kmeans.py#L186) doesn't use `whiten` as far as I can tell.

What I haven't looked at in detail yet is if it would help the performance of `kmeans` to do the actual whitening, or if we're good with rescaling for equal variances.

**Comment by josef-pkt** [comment]
The most common usage of "standardization" includes demeaning and not just rescaling.
I don't know what vq whiten does.

`whiten` in statsmodels usage does not remove the mean.
